Update of landuse outlines

dch0ph opened the pull request:
Fixes #5083

Changes proposed in this pull request:

- Remove dependence of outline width on presence of name. Currently the outline for most "larger" landuses with a name is widened from 0.5 px to 0.7 px. This is barely visible and not providing useful mapper feedback. All these outlines now have a common width of `landuse-outline-width` (set to 0.5 px).
- There was a mistake / inconsistency in the `polygon-gamma` filters in `landuse_retail`.
- There was a inconsistency in `landuse=railway` whereby an outline was only shown at all from Z16 if the feature is named. This leads to the only visible change from this PR.
- There was a redundant filter in `man_made=wastewater_plant` and `man_made=water_works` to switch fill colour from `@industrial` to `@wastewater_plant` at Z15+. Since these two colours are the same (and it wouldn't make sense to switch a fill colour at Z15+ anyway), these filters are redundant. 

Test rendering with links to the example places:

[Example](https://www.openstreetmap.org/edit#map=17/54.803990/-1.582015) showing that the change from 0.7 px to 0.5 px for named landuses (retail area and residential area) is not really visible:

Before
![image](https://github.com/user-attachments/assets/c6a246dd-9499-4205-9073-6d43ff4da603)

After
![image](https://github.com/user-attachments/assets/b0c5b031-3bad-4bb6-889e-fdc8a742ed72)

[Example](https://www.openstreetmap.org/edit#map=17/54.766134/-1.605490) of outline on un-named `landuse=railway`: 

Before
![image](https://github.com/user-attachments/assets/cc01ac17-5230-4df2-bc65-0a7d909503b9)

After
![image](https://github.com/user-attachments/assets/05937cf1-9c18-45d3-880d-94ebcf258333)

Queries:

- It might make sense to use an intermediate linewidth i.e. 0.6 px. As it stands, the PR takes a conservative line by using 0.5 px, which will have no effect on the majority of cases.
- It seems odd to have separate MSS code for `man_made=wastewater_plant` and `man_made=water_works` since these are rendered in exactly the same way.
- The starting zooms for outline rendering vary wildly. `leisure=ice_rink` renders a 0.5 px outline from Z10 (really?!), while most outlines considered in this PR start at Z16 (which seems a little late, Z15 better?).

imagico commented:
> It might make sense to use an intermediate linewidth i.e. 0.6 px. As it stands, the PR takes a conservative line by using 0.5 px, which will have no effect on the majority of cases.

Since the difference in appearance between landcover outlines and line barriers is already often subtle i would be careful here.

> It seems odd to have separate MSS code for man_made=wastewater_plant and man_made=water_works since these are rendered in exactly the same way.

That was probably meant to allow derivative styles to easily use different colors - for the goal of adaptability.

> The starting zooms for outline rendering vary wildly.

There are different types of landcover polygons - the urban landcovers, for example, are frequently split in mapping at relatively arbitrary locations.  Here the purpose of the outline is to provide a subtle feedback on where the split is without making it appear as something that is necessarily semantically meaningful.  In other cases - like with `leisure=playground` or the mentioned ice rinks the polygon is commonly understood to always represent a distinct feature.

The `name` tag dependent styling was meant to distinguish between these two cases for the same primary tag.  But this is often not a very meaningful indicator.  For `landuse=railway` the idea probably was that this tag even less frequently represents a distinct feature than other landuse tags - hence the outline is only shown when a name tag is present.

We make the distinction between landcovers rendered with an outline and those which are not and we are not going to change that.  Hence it needs to be decided on a case-by-case basis where to make this distinction.

Outline rendering of the landcover polygons that are frequently split arbitrarily should not start earlier than line barrier rendering.

The following design concepts are currently in use and make at least some sense:

* always with outline starting with buildings (z15) - makes sense for things like `leisure=pitch`
* subtle outline later (z16) - makes sense for things like `landuse=farmland` where the polygons frequently represent distinct units with distinct characteristics.
* subtle outline only with presence of other tags - currently for `landuse=railway`, depends on if the other tags are actually a meaningful indicator for the class in question.
* no outline - makes sense for most vegetation related landcovers where the individual polygon rarely represents a distinct feature.

dch0ph commented:
Many thanks for these clarifications.

In terms of `landuse=railway`, I should have found the discussion in the original PR (#599), in which there there was agreement that an outline on `landuse=railway` was not generally useful, and retaining it for the case of named areas seems to have been a compromise so that the boundary of these areas could be seen. We don't generally do this in other cases, e.g. named sections of woodland. Majority usage of named `landuse=railway` seems to be for junctions e.g. [here](https://www.openstreetmap.org/edit#map=18/54.991137/-1.562556):

![image](https://github.com/user-attachments/assets/478ce77f-c0be-4669-8364-4e9890d21c27)

Note the thin outline associated with the naming in the bottom left.

I think these boundaries are pretty arbitrary and mappers are focussing on labelling the junctions, sidings yards etc. rather than distinguishing the boundaries with respect to other railway areas.

The two outliers from the general design principles above are:

- `leisure=ice_rink` where the outline starts at Z10, rather than Z13 (which is the earliest start zoom for all other outlines). This doesn't seem to have been considered in the PR discussion (#3330).
- `leisure=swimming_pool` where the outline was pushed back to Z17 (commit doesn't seem part of a discussion), perhaps because most outlines start 3 zoom levels after the area render (which starts at Z14 for swimming pools). I don't think this late start makes sense, not least because you then can't distinguish between a swimming pool and an area of water.

[Example](https://www.openstreetmap.org/edit#map=16/54.74881/-2.01053) at Z16:

![image](https://github.com/user-attachments/assets/7803eca4-728f-412f-bae4-e2d625289be3)

It would make more sense to start at Z15, as `leisure=pitch` etc.

imagico commented:
Please keep in mind that both `leisure=ice_rink` and `leisure=swimming_pool` are rendered in a design where the outline is the only differentiation from other features that are fundamentally different in meaning.

The main difference is that in the latter case that difference is harmless - hence it is reasonable to render them just like a generic water area at scales where the outline would not be well readable.  In the former case that is different.  Here the outline was added early on in an attempt to disguise the poor choice of fill color - conflating features that are semantically not related at all.  IMO it would make more sense to render `leisure=ice_rink` like `leisure=pitch` with additional differentiation via symbol (#844).

Also should take into account the typical sizes of the features in question - `leisure=pitch` on average are much larger than `leisure=swimming_pool`.  And open air swimming pools are mostly limited to lower latitudes while open air ice rinks mostly exist at high latitudes.

> I think these boundaries are pretty arbitrary and mappers are focussing on labelling the junctions, sidings yards etc. rather than distinguishing the boundaries with respect to other railway areas.

The rendering of name labels on plain `landuse=railway` without further qualifying tags (`man_made=*`, `railway=*`) is another matter.

dch0ph commented:
Changes made based on discussion:
- Documented design logic of "faint outlines" and removed `landuse=quarry` from this category, i.e. `@landuse-faint-outline-width` is only used for renders starting at Z16 (which coincides with point at which most barriers are rendered).
- Removed outlining for `landuse=railway`, so the practical effect of this PR is limited to removing the outlining from named `landuse=railway`. Hence [example above](https://www.openstreetmap.org/edit#map=18/54.991137/-1.562556) becomes:

![image](https://github.com/user-attachments/assets/efc55e26-00b2-4ad2-856f-7d151c053f8c)

(No faint line in bottom left)

[On `leisure=ice_rink`]
>  Here the outline was added early on in an attempt to disguise the poor choice of fill color - conflating features that are semantically not related at all.

OK. Let's leave this. I just assumed that we wouldn't want to render outlines at low zoom as they would just be visual noise.
